gnomad_qc/v4/annotations/generate_freq.py

```diff
@@ -0,0 +1,441 @@
+"""Script to generate the frequency data annotations across v4 exomes."""  # TODO: Expand on script description once nearing completion.
+import argparse
+import logging
+
+import hail as hl
+from gnomad.resources.grch38.gnomad import (
+    SUBSETS,  # TODO: subsets will be changed to UKB, non-UKB, non-TopMed
+)
+from gnomad.resources.grch38.gnomad import (
+    DOWNSAMPLINGS,
+    POPS,
+    POPS_TO_REMOVE_FOR_POPMAX,
+)
+from gnomad.sample_qc.sex import adjusted_sex_ploidy_expr
+from gnomad.utils.annotations import (
+    age_hists_expr,
+    annotate_freq,
+    bi_allelic_site_inbreeding_expr,
+    faf_expr,
+    get_adj_expr,
+    pop_max_expr,
+    qual_hist_expr,
+    set_female_y_metrics_to_na_expr,
+)
+from gnomad.utils.file_utils import file_exists
+from gnomad.utils.release import make_faf_index_dict, make_freq_index_dict
+from gnomad.utils.slack import slack_notifications
+
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v4.resources.annotations import (  # TODO: Need to generate this resource for raw callstats
+    get_freq,
+)
+from gnomad_qc.v4.resources.basics import (
+    get_checkpoint_path,
+    get_gnomad_v4_vds,
+    qc_temp_prefix,
+)
+from gnomad_qc.v4.resources.meta import meta
+from gnomad_qc.v4.resources.release import release_sites
+from gnomad_qc.v4.resources.variant_qc import (  # TODO: Confirm we want this in frequency calculations
+    SYNDIP,
+)
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s: %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("gnomAD_frequency")
+logger.setLevel(logging.INFO)
+
+
+def annotate_non_ref_het(vds: hl.vds.VariantDataset) -> hl.vds.VariantDataset:
+    """
+    Annotate non-ref heterozygous calls to prevent incorrect call adjustment post-split.
+
+    :param vds: Hail VDS to annotate non-ref het onto variant data.
+    :return: Hail VDS with _non_ref_het annotation.
+    """
+    vds.variant_data = vds.variant_data.annotate_entries(
+        _het_non_ref=vds.variant_data.LGT.is_het_non_ref()
+    )
```

Comment:
Nice, I didn't realize this worked, I have always used something like:

```
vmt = vds.variant_data
vmt = vmt.annotate_entries(_het_non_ref=vmt.LGT.is_het_non_ref())
vds = hl.vds.VariantDataset(vds.reference_data, vmt)
```

good to know